collector/src/execute.rs

```diff
@@ -1239,6 +1256,13 @@ impl Benchmark {
         self.config.supports_stable
     }
 
+    pub fn category_name(&self) -> &str {
```

Comment:
Make this a method on `Category` itself? (Possibly even `impl Display`) Then below where you currently have `"primary"` you could instead do `(Category::Primary).to_string()`